drud/ddev-ui
# Better Error handling and alert management #141

## The Problem/Issue/Bug:
This PR addresses #141 

## How this PR Solves The Problem:
Adds React error boundaries. Also reworks the Alerts component using toastify notifications framework.

## Manual Testing Instructions:
Download new build - (https://417-99947022-gh.circle-artifacts.com/0/artifacts/DDEV%20UI-0.4.2-alpha.dmg)
1. Open app without Docker open.
2. Open docker While App is open to see notice go away.
3. Open app with docker open.
4. Notice no errors are shown.
5. Stop docker while the app is open.
6. All the rest of the app should continue to work as normal.

## Automated Testing Overview:
Still disabled 🙈

## Related Issue Link(s):
#141 

## Release/Deployment notes:
- Provided more intuitive and in-depth error catching and handling.
- Reworked the alerts component using toastify
- Docker errors will trigger the system check again
